docs/assets.md

```diff
@@ -0,0 +1,151 @@
+
+![Featured Image Upload Flow](diagrams/asset_featured_image_upload_flow.png)
+
+****
+
+# Cross functional requirements for File Attachments
+
+*FileAttachments are considered to be important data for the document quality. They can contain detailed 
+information which is crucial for the containing document.*
+
+
+**Correctness** over **Maintainability**
+
+> An attachment can be configured to redirect the user to a newer version of the attachment.
+This helps ensure, anyone who follows an outdated url - will be given correct up-to-date
+information. This comes at the cost of complexity in maintaining redirect-urls for attachments.
+
+**Obtainability** over **code simplicity**
+
+> An attachment can be replaced by another attachment. These replacements are tracked in the system
+so that potentially the user can be provided with always up to date version of the information. 
+
+**Recoverability** over **code simplicity**  
+
+> Attachments are always soft-deleted,
+this makes it possible to recover the attachments both in Whitehall and
+Asset Manager when needed. The cost of soft-deletion is code that is harder to
+maintain and understand.
+
+
+# Cross functional requirements for Images
+
+*Images are considered mainly decorative and visually supportive items
+for a document. They should not contain any crucial information
+in order for the document to be understood.*
+
+**Accessibility**
+
+> People who use screen readers should be able to gain just as much information
+of the document at hand - whether they can see the image or not. Alt texts are always available
+to provide descriptive information of an image.
+
+**Code simplicity** over **Recoverability** 
+
+> Images are not important information. They are generally hard-deleted from Whitehall
+database in favor of having simple and readable code.
+
+**User experience** over **Maintainability**
+
+> Different devices require different image sizes in order for the ui to look good.
+These image variants need to be created and the code for it needs support and maintenance.
+
+
+# Common cross functional requirements
+
+**Security and Safety** over **ease of support**
+
+> All attachments and images are scanned for potential malware in Asset Manager. This
+is done to ensure the safety of the users of gov.uk. The cost of this is that sometimes
+2nd line support people have to fix attachments that have false positives from the
+virus scanner.
+
+**User experience** over **code simplicity**
+
+> File uploads are managed internally with Asynchronous background jobs. This helps the publisher-user
+have a fast user experience when uploading files. However it comes at the cost of complicated code flows
+that manage the successes and failures of the file uploads.
+
+# Business Requirements
+
+#### *Members of public* need to be see appropriate image sizes depending on the device they are using
+
+> Whitehall can publish several image urls as per necessary for various sizes. 
```

Comment:
english is not my greatest strength, so I could be tottaly wrong here..
is as per necessary  = as per necessity is what we meant here?